### PR TITLE
Batch cutlass distance kernels along N matrix dim

### PR DESCRIPTION
**Pull request by mdoijade**
- batch input matrix with N-dim to at most 65535 to avoid cutlass gridY limitation.

### CODE REVIEW

**Review by tfeher — CHANGES_REQUESTED** [review]
Thanks Mahesh for fixing the issue! The PR looks good to me, but could you still add a test?

File: cpp/include/raft/distance/detail/pairwise_distance_cutlass_base.cuh [hunk 1]
```diff
@@ -116,53 +109,73 @@ std::enable_if_t<ops::has_cutlass_op<OpT>::value> cutlassDistanceKernel(const Da
...
-    b        = x;
-    gemm_lda = ldb;
-    gemm_ldb = lda;
-  } else {
-    problem_size = cutlass::gemm::GemmCoord(m, n, k);
-    a            = x;
-    b            = y;
-    gemm_lda     = lda;
-    gemm_ldb     = ldb;
+  constexpr uint32_t gridYZMax      = ((1 << (sizeof(uint16_t) * 8)) - 1);
```
[the remark this answers is not in the document]
Comment (mdoijade) [reply cpp/include/raft/distance/detail/pairwise_distance_cutlass_base.cuh:112]: It is taken from cutlass code, this is how they decide limit. We can certainly replace it with `65535` if you are in favor for that.

[the remark this answers is not in the document]
Comment (tfeher) [reply cpp/include/raft/distance/detail/pairwise_distance_cutlass_base.cuh:112]: We can keep it to be consistent with cutlass.

**Review by tfeher — APPROVED** [review]
Thanks Mahesh for the update, LGTM!

**Pull request merged**